src/develop/pixelpipe_cache.c

```diff
@@ -213,84 +212,83 @@ static int _get_oldest_cacheline(dt_dev_pixelpipe_cache_t *cache)
 static int _get_free_cacheline(dt_dev_pixelpipe_cache_t *cache, size_t size)
 {
   int oldest = _get_oldest_cacheline(cache);
-  if((cache->memlimit == 0) || ((cache->memlimit - cache->allmem) >= size))
+  if((cache->memlimit == 0) || (cache->memlimit > cache->allmem))
     return oldest;
 
-  int cnt = 0;
-   while(((cache->memlimit - cache->allmem) < size) && (cnt < cache->entries))
+  // we have to free & invalidate cachelines until there is enough mem available
+  size_t freed = 0;
+  oldest = _get_oldest_used_cacheline(cache);
+  while((cache->memlimit < cache->allmem) && (oldest >= 0))
   {
-    cnt++;
-    // we have to free & invalidate cachelines until there is enough mem available
     dt_free_align(cache->data[oldest]);
     cache->allmem -= cache->size[oldest];
- 
+    freed += cache->size[oldest];
     cache->size[oldest] = 0;
     cache->data[oldest] = NULL;
     cache->hash[oldest] = -1;
     cache->basichash[oldest] = -1;
-    cache->used[oldest] = 1;
-    oldest = _get_oldest_cacheline(cache);
+    cache->used[oldest] = 1000;
```

Comment:
We have 1000 used in 3 more places. IIUC this is an "old" age for the entry. Probably using a #define (a named number) for this would make the code easier to read.

Reply:
good point